lnurl-allowance.md

```diff
@@ -0,0 +1,197 @@
+# LNURL-allowance
+
+The LNURL Allowance scheme allows a `LN SERVICE` and a `LN WALLET` to establish a `pairing` over a WebSocket connection. The idea is that `LN SERVICE` will provide an LNURL that returns some parameters (including a socket URL) to `LN WALLET`. `LN WALLET` will diplay a pairing dialog and once accepted by the user, `LN WALLET` will start a socket connection to `LN SERVICE`. This will allow for `LN WALLET` to pay/receive funds to/from `LN SERVICE` without user interaction or QR Code scanning, up to the established allowance amount.
+
+## Scan QR code to establish allowance/pairing
+
+### WALLET to SERVICE interaction flow:
+
+1. User scans a LNURL QR code or accesses a `lightning:LNURL..` link with `LN WALLET` and `LN WALLET` decodes LNURL.
+
+2. `LN WALLET` makes a GET request to `LN SERVICE` using the decoded LNURL.
+
+3. `LN WALLET` gets JSON response from `LN SERVICE` of form:
+
+```
+{
+    socket: String, // the URL from LN SERVICE which will accept the WebSocket connections from the WALLET
+    k1: String, // random or non-random string to identify the WALLET when receiving connection to socket
+    recommendedAllowanceAmount: Millsatoshis, // amount of satoshis the SERVICE recommends WALLET to ask user for this allowance/pairing
+    image: Base64, // base64-encoded LN SERVICE image for LN WALLET to display to user when establishing connection
+    description: String, // LN SERVICE description that LN WALLET should showcase to user
```

Comment:
should this field be more like lnurl-auth rather than lnurl-withdraw? 
Maybe what's shown to the user needs to be pulled from domain to ensure the request comes from the actual website. 
otherwise anyone can put a description with a popular game and immediately withdraw all the allowance.

Reply:
Yes I agree with this. As you can see, the WALLET is responsible for doing exactly what you mentioned: [you can see that here](https://github.com/btcontract/lnurl-rfc/pull/49/files/528191fe8c39036e12c6e9605791178704698d7c#diff-0a4dd2f0bb49676afea53bb019139103R40)